server/Server.js

```diff
@@ -323,6 +327,62 @@ class Server {
     }
   }
 
+  // Apply the current RouterBasePath to all `<base>` tags in the client html files
+  async applyClientBasePath () {
+    const basePath = `/${global.RouterBasePath}/`
+      .replace('///', '') // convert base path `"/""` to `""`
+      .replace('//', '/') // Ensure there are single `/` at the front and end
+      .slice(0, -1) // Remove trailing `/`
+    const basePathSlash = `${basePath}/`
+    const distPath = Path.join(global.appRoot, '/client/dist')
+
+    // Check if the client base path is already set to the current base path
+    const indexHtml = await fs.readFile(Path.join(distPath, 'index.html'), 'utf8')
+    const currentBasePath = indexHtml.match(/routerBasePath:"([^"]*)"/)[1]
+    if (currentBasePath === basePath) {
+      Logger.info(`[Server] Client base path already set to ${basePath}`)
+      return;
+    }
+
+    // Matches e.g `href="/_nuxt/..."` or `src='/...'` or `"url":"http://localhost:3333/...`
+    const replaceUrls = new RegExp(`((?:href|src|url|scope)"?(?:=|:)\\s*["'](?:http://localhost:3333)?)${currentBasePath || "/"}([^"']*["'])`, "g")
+    const replaceNuxtConfig = new RegExp(`<script>window.__NUXT__=.*?</script>`, "g")
+    const replaceAssetPaths = new RegExp(`${currentBasePath}/(_nuxt/[^"]*|sw.js)`, "g")
```

Comment:
I looked through every single change this made, and they all looked fine. No guarantee though that it will stay this way, nor that it will always catch every path to be updated.